dist/rpm/os-autoinst.spec

```diff
@@ -28,6 +28,7 @@ Source0:        %{name}-%{version}.tar.xz
 %if 0%{?suse_version} > 1500
 # openSUSE Tumbleweed
 %define opencv_require pkgconfig(opencv4)
+BuildRequires:  swtpm
```

Comment:
This makes it look like only on Tumbleweed we would actually require the package and we don't need the package anywhere else while still delivering the same functionality which is simply not true. I think we can do better than this. I am thinking, seeing on https://build.opensuse.org/search?search_text=swtpm that swtpm is available in both Tumbleweed as well as SLE15-SP3 aka. Leap 15.3 I see the following options:
1. simply hard-require the package and drop support for Leap 15.2 and others that don't provide swtpm
2. same as 1 but we try to build swtpm ourselves, e.g. in devel:openQA:Leap:15.2 . This might be tricky however as also in https://build.opensuse.org/package/show/security/swtpm the package is unresolvable on Leap 15.2
3. As suggested in before don't require it and rewrite the test to not rely on the package itself but mock the dependency and keep the convenience package only
4. Turn the "convenience" package into a proper package with the swtpm only provided in that package, e.g. by extracting the functionality into a dedicated child-backend like "qemu-swtpm" and deliver the according file(s) only in the according package which requires swtpm

I still prefer 3., next to 1., but let's others have their opinion heard as well

Reply:
This is only relevant for tests.

Reply:
I assume `BuildRequires` relates to the build host. We also uses that for `%test_requires`

Reply:
I know that. But I don't like to re-introduce special test-dependency handling into the specfile. We have dependencies.yaml so we should use it

Reply:
I don't see how `dependencies.yaml` can be used here, and I don't see how any of your suggestions helps with what I intended to do. That's why I thought you didn't understand it.

Reply:
Tests seem to pass unmodified without `swtpm` installed so I'll leave it at that.